src/editors/selectEditor.js

```diff
@@ -15,233 +15,288 @@ import { KEY_CODES } from './../helpers/unicode';
-  this.instance.addHook('afterRowResize', () => this.refreshDimensions());
-};
-
-SelectEditor.prototype.prepare = function(...args) {
-  BaseEditor.prototype.prepare.apply(this, args);
-
-  const selectOptions = this.cellProperties.selectOptions;
-  let options;
-
-  if (typeof selectOptions === 'function') {
-    options = this.prepareOptions(selectOptions(this.row, this.col, this.prop));
-  } else {
-    options = this.prepareOptions(selectOptions);
+class SelectEditor extends BaseEditor {
+  /**
+   * Initializes editor instance, DOM Element and mount hooks.
+   */
+  init() {
+    this.select = document.createElement('SELECT');
+    addClass(this.select, 'htSelectEditor');
+    this.select.style.display = 'none';
+    this.instance.rootElement.appendChild(this.select);
+    this.registerHooks();
   }
 
-  empty(this.select);
+  /**
+   * Returns select's value.
+   *
+   * @returns {*}
+   */
+  getValue() {
+    return this.select.value;
+  }
 
-  objectEach(options, (value, key) => {
-    const optionElement = document.createElement('OPTION');
-    optionElement.value = key;
-    fastInnerHTML(optionElement, value);
-    this.select.appendChild(optionElement);
-  });
-};
+  /**
+   * Sets value in the select element.
+   *
+   * @param {*} value A new select's value.
+   */
+  setValue(value) {
+    this.select.value = value;
+  }
 
-SelectEditor.prototype.prepareOptions = function(optionsToPrepare) {
-  let preparedOptions = {};
+  /**
+   * Opens the editor and adjust size.
+   */
+  open() {
+    this._opened = true;
+    this.refreshDimensions();
+    this.select.style.display = '';
+    this.instance.addHook('beforeKeyDown', () => this.onBeforeKeyDown());
+  }
 
-  if (Array.isArray(optionsToPrepare)) {
-    for (let i = 0, len = optionsToPrepare.length; i < len; i++) {
-      preparedOptions[optionsToPrepare[i]] = optionsToPrepare[i];
-    }
-  } else if (typeof optionsToPrepare === 'object') {
-    preparedOptions = optionsToPrepare;
+  /**
+   * Closes the editor.
+   */
+  close() {
+    this._opened = false;
+    this.select.style.display = 'none';
+    this.instance.removeHook('beforeKeyDown', () => this.onBeforeKeyDown());
   }
 
-  return preparedOptions;
+  /**
+   * Sets focus state on the select element.
+   */
+  focus() {
+    this.select.focus();
+  }
 
-};
+  /**
+   * Binds hooks to refresh editor's size after scrolling of the viewport or resizing of columns/rows.
+   *
+   * @private
+   */
+  registerHooks() {
+    this.instance.addHook('afterScrollHorizontally', () => this.refreshDimensions());
+    this.instance.addHook('afterScrollVertically', () => this.refreshDimensions());
+    this.instance.addHook('afterColumnResize', () => this.refreshDimensions());
+    this.instance.addHook('afterRowResize', () => this.refreshDimensions());
+  }
 
-SelectEditor.prototype.getValue = function() {
-  return this.select.value;
-};
+  /**
+   * Prepares a list of available options.
+   */
+  prepare(...args) {
```

Comment:
Shouldn't this method be private?
Also, maybe it would be a good idea to add JSDoc to the args that could be passed to it?

Reply:
I think it could be public as in `_baseEditor`. That method prepares the editor's meta-data also.
Yeah, JSDoc to the args will be helpful :+1: